source/ai/rl/BBF_agent/BBF.py

```
@@ -0,0 +1,442 @@
+                )
+                n = np.array(n).item()
+
+                self.memory.priority[len_memory] = self.memory.max_priority()
+
+                if len_memory > 2000:
+                    for i in range(self.replay_ratio):
+                        self.optimize(grad_step, n)
+                        target_model_ema(self.model, self.model_target)
+                        grad_step += 1
+
+                if save_freq != None and ((step + 1) % save_freq) == 0:
```

Ruff (0.12.2)

Comparison to `None` should be `cond is not None`

Replace with `cond is not None`

(E711)